src/App.js

```diff
@@ -0,0 +1,201 @@
+import React, { Component } from "react";
+
+import TodoItemsList from "./components/TodoItemsList";
+import TodoInput from "./components/TodoInput";
+import TodoSearch from "./components/TodoSearch";
+import "./App.css";
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "",
+      surname: "",
+      message: "",
+      isSubmitted: false,
+      key: 2,
+      todoItems: [
+        {
+          name: "Andrew",
+          surname: "anime",
+          message: "Hello World!",
+          date: '12.10.2018',
+          key: 1,
+          isCompleted: false,
+          display: true,
+          textMatchStart: "",
+          textMatch: false,
+          textMatchEnd: ""
+        },
+        {
+          name: "Yura",
+          surname: "notanime",
+          message: "Hello World!",
+          date: '12.10.2018',
+          key: 2,
+          isCompleted: true,
+          display: true,
+          textMatchStart: "",
+          textMatch: false,
+          textMatchEnd: ""
+        }
+      ]
+    };
+  }
+
+  handleChange = e => {
+    this.setState({ [e.target.name]: e.target.value })
+  };
+
+  handleAdd = e => {
+    let newKey = this.state.key;
+    newKey++;
+    this.setState({ isSubmitted: true, key: newKey });
+    this.setState({
+      todoItems: [
+        {
+          name: this.state.name,
+          surname: this.state.surname,
+          message: this.state.message,
+          date: this.getCurDate(),
+          key: newKey,
+          isCompleted: false,
+          display: true,
+          textMatchStart: "",
+          textMatch: false,
+          textMatchEnd: ""
+        },
+        ...this.state.todoItems
+      ]
+    });
+  };
+
+  validateInput = () => {
+    if (this.state.name && this.state.surname && this.state.message) {
+      console.log('true');
+      return true;
+    }
+    console.log('false');
+    return false;
+  }
+
+  getCurDate = () => {
+    var today = new Date();
+    return today = `${(today.getDate() < 10) ? '0' + today.getDate() : today.getDate()}.${(today.getMonth() < 10) ? '0' + today.getMonth() : today.getMonth()}.${today.getFullYear()}`;
+  }
+
+  toggleComplete = index => {
+    let arrCopy = [...this.state.todoItems];
+    arrCopy[index].isCompleted = !arrCopy[index].isCompleted;
+    this.setState({
+      todoItems: arrCopy
+    });
+  };
+
+  toggleDelete = (inputKey) => {
+    let arrCopy = [...this.state.todoItems];
+    console.log(inputKey);
+    this.setState({
+      todoItems: arrCopy.filter(
+        // cur => cur.name !== inputName && cur.surname !== inputSurname
+        cur => cur.key !== inputKey
+      )
+    });
+  };
+
+  handleSearch = e => {
```

Comment:
I'm not really sure what is done here inside `handleSearch` but `handleSearch` should only set a `searchValue` on a state, then you need to do `filter` items based on the `searchValue` inside render function.

e.g.

```
handleSearch = e => this.setState({ searchValue: e.target.value })
```

```
render() {
  const filteredItems = this.state.allItems.filter(item => item.name.includes(this.state.searchValue));

  return filteredItems.map(item => <li>{item.name}</li>)
}
```

some example on official docs https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization

Let me know if it is clear for you.